lib/ABI/FunctionType.cpp

```diff
@@ -338,13 +377,14 @@ class ConversionHelper {
         } else if (Result.size() > 1 && Index > 1) {
           auto &First = Result[Result.size() - 1];
           auto &Second = Result[Result.size() - 2];
-          if (!First.CustomName.empty() || !Second.CustomName.empty()) {
-            if (First.CustomName.empty())
-              First.CustomName = "unnamed";
-            if (Second.CustomName.empty())
-              Second.CustomName = "unnamed";
-            First.CustomName.append(("+" + Second.CustomName).str());
-          }
+
+          if (First.name().empty())
+            First.CustomName = "unnamed";
+          if (Second.name().empty())
+            Second.CustomName = "unnamed";
+          First.CustomName = ("combination_of_" + First.name()).str();
+          First.CustomName.append(("_and_" + Second.name()).str());
```

Comment:
We never generate any name.

Reply:
`First.OriginalName` doesn't work either?

Reply:
Here it's a bit more tricky that the other way - here we have two different arguments - there's no way to determine which one of the two to discard.

Reply:
If we think this is very important, we can turn `OriginalName` into a vector one day. Feel free to put a TODO, and if we see a surging demand for something like this, we can address it.

Reply:
Done: https://github.com/revng/revng/pull/148/commits/91b7d815be9d8bbf0a526577804567ddb991e229.